Library/Homebrew/cmd/test-bot.rb

```diff
@@ -547,35 +547,41 @@ def formula(formula_name)
         test "brew", "postinstall", *changed_dependences
       end
       formula_fetch_options = []
-      formula_fetch_options << "--build-bottle" unless ARGV.include? "--no-bottle"
       formula_fetch_options << "--force" if ARGV.include? "--cleanup"
       formula_fetch_options << canonical_formula_name
       test "brew", "fetch", "--retry", *formula_fetch_options
       test "brew", "uninstall", "--force", canonical_formula_name if formula.installed?
       install_args = ["--verbose"]
-      install_args << "--build-bottle" unless ARGV.include? "--no-bottle"
       install_args << "--HEAD" if ARGV.include? "--HEAD"
 
       # Pass --devel or --HEAD to install in the event formulae lack stable. Supports devel-only/head-only.
       # head-only should not have devel, but devel-only can have head. Stable can have all three.
       if devel_only_tap? formula
         install_args << "--devel"
+        formula_bottled = false
       elsif head_only_tap? formula
         install_args << "--HEAD"
+        formula_bottled = false
+      else
+        formula_bottled = formula.bottled?
       end
 
       install_args << canonical_formula_name
       # Don't care about e.g. bottle failures for dependencies.
       run_as_not_developer do
-        test "brew", "install", "--only-dependencies", *install_args unless dependencies.empty?
-        test "brew", "install", *install_args
+        if !ARGV.include?("--fast") || formula_bottled
+          test "brew", "install", "--only-dependencies", *install_args unless dependencies.empty?
+          test "brew", "install", *install_args
+          install_passed = steps.last.passed?
+        else
+          install_passed = false
+        end
       end
-      install_passed = steps.last.passed?
```

Comment:
Just caught this: moving `install_passed` inside the `run_as_not_developer` block broke the `if install_passed` check; the variable is not defined anymore outside of the block. Fixed in a6611f9f.